pkg/minishift/registration/redhat.go

```diff
@@ -78,3 +80,26 @@ func (registrator *RedHatRegistrator) Unregister(param *RegistrationParameters)
 	}
 	return nil
 }
+
+//Put required proxy information to subscription-manager config file
+func (registrator *RedHatRegistrator) RegistrationManagerConfig(proxyInfo string) error {
+	if matched, _ := regexp.MatchString("http://", proxyInfo); matched {
+		proxyInfo = strings.TrimPrefix(proxyInfo, "http://")
+	} else if matched, _ := regexp.MatchString("https://", proxyInfo); matched {
+		fmt.Println("Removing https")
+		proxyInfo = strings.TrimPrefix(proxyInfo, "https://")
+	}
+
+	//Value of proxyInfo at this time would be similar to fedora:fedora123@xyz.com:8000
+	//Replacing the "@" with ":" and then splitting it with ":" will keep required values in a single slice
+	proxyInfo = strings.Replace(proxyInfo, "@", ":", 1)
```

Comment:
> I realize that it is tricky. See [1] . However if we take single argument for proxy it is always going to be like this. May be we can add documentation about it.

What do you mean? Having an '@' this is a valid case, we cannot control the password. We need to be smart enough here to parse this URL in a way where this is possible. I don't even think it needs to be escaped. You just need to write the parsing algorithm in a different way.